src/protocol/record.rs

```diff
@@ -592,6 +594,42 @@ where
 
                 records
             }
+            #[cfg(feature = "compression-snappy")]
+            RecordBatchCompression::Snappy => {
+                use snap::raw::{decompress_len, Decoder};
+
+                // Construct the input for the raw decoder.
+                let mut input = vec![];
+                reader.read_to_end(&mut input)?;
```

Comment:
Would this allow a malicious input to trigger unbounded memory usage by feeding it `/dev/rand` for a bit?

Reply:
Not here because the reader is capped (it's reading from an in-memory vector that we already have at this point and this one is limited by `max_message_size` which is a protocol/client-level config). However now that I think about it we don't limit the "uncompressed size" and snappy doesn't limit that one. I'll fix that attack vector.

Reply:
fixed